File: 문자열/9251 - LCS.py
import sys
read = sys.stdin.readline

s1 = read().strip()
s2 = read().strip()

graph = [[0] * (len(s2)+1) for _ in range(len(s1)+1)]

for i in range(len(s1)):
    for j in range(len(s2)):
        if s1[i] == s2[j]:
            graph[i+1][j+1] = graph[i][j]+1
        else:
            graph[i+1][j+1] = max(graph[i+1][j], graph[i][j+1])

print(graph[-1][-1])

File: 문자열/9251 - LCS.js
const fs = require('fs');
const input = fs
  .readFileSync('./문자열/9251input.txt')
  .toString()
  .trim()
  .split('\r\n');
// const input = fs.readFileSync('dev/stdin').toString().trim().split('\n');

const s1 = input[0];
const s2 = input[1];

const l1 = s1.length;
const l2 = s2.length;

const graph = Array.from(Array(l1 + 1), () => Array(l2 + 1).fill(0));

for (let i = 0; i < l1; i++) {
  for (let j = 0; j < l2; j++) {
    if (s1[i] === s2[j]) {
      graph[i + 1][j + 1] = graph[i][j] + 1;
    } else {
      graph[i + 1][j + 1] = Math.max(graph[i + 1][j], graph[i][j + 1]);
    }
  }
}

console.log(graph[l1][l2]);
